enterprise/server/util/redisutil/redisutil.go

```diff
@@ -68,43 +68,30 @@ func TargetToOptions(redisTarget string) *redis.Options {
 	}
 }
 
-func TargetToOpts(redisTarget string, useTLS bool) *Opts {
+func TargetToOpts(redisTarget string) *Opts {
 	if redisTarget == "" {
 		return nil
 	}
 	libOpts := TargetToOptions(redisTarget)
-	var tlsConfig *tls.Config
-	if useTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
-		}
-	}
 	return &Opts{
 		Addrs:     []string{libOpts.Addr},
 		Network:   libOpts.Network,
 		Username:  libOpts.Username,
 		Password:  libOpts.Password,
 		DB:        libOpts.DB,
-		TLSConfig: tlsConfig,
+		TLSConfig: libOpts.TLSConfig,
 	}
 }
 
-func ShardsToOpts(shards []string, useTLS bool, username, password string) *Opts {
+func ShardsToOpts(shards []string, username, password string) *Opts {
 	if len(shards) == 0 {
 		return nil
 	}
-	var tlsConfig *tls.Config
-	if useTLS {
-		tlsConfig = &tls.Config{
-			MinVersion: tls.VersionTLS12,
-		}
-	}
-	return &Opts{
-		Addrs:     shards,
-		Username:  username,
-		Password:  password,
-		TLSConfig: tlsConfig,
-	}
+	opt := TargetToOpts(shards[0])
```

Comment:
nit: could we return an error here if the shards are using a mix of `rediss://` and `redis://` ?

Reply:
our current code to setup redis does not support returning an error 🤔 

We can log a warning return nil, in which case the non-sharded config will get picked up. If there is no default then there will be no redis client registered and we fail to startup in case of enterprise setup.

We can also do log.fatal directly here, which is what the error return gona do regardless as this failed during registration.

I am leaning toward just log a warning and returning nil, wdyt?

Reply:
If it's too big of a refactor to return an error then logging a warning seems fine as well